common/src/ProductInfo.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Devices.Common
+{
+    internal class ProductInfo
```

Comment:
Missing test. Given the current implementation for DeviceClient, I'm not sure this actually works as expected.
We have one within DeviceClient which has a unit test. If you decide to move that into common (although Name would always be different), please ensure that everything else that's using ProductInfo uses the same implementation (DeviceClient, ProvisioningClient).

Reply:
GetExecutingAssembly() is not available in netstandard1.3. So for now, the ProductInfo in common can only be used by the ProvisioningDevoceClient and ProvisioningServiceClient. I will added unit tests for this separately.

Reply:
I left it in common as it is now used by both provisioning device client and provisioning service client. I also had to rearrange the "common" folder structure so that it was aligned with our project folder structure.